### PR TITLE
Addonchecker: Added diagnostics and GMOD version check

### PR DESCRIPTION
**Pull request by NickCloudAT**
This adds a bit more information to the end of the addonchecker for diagnostic purpose.
It also contains a Garry's Mod Version check for the minimum required version which TTT2 can run on.

Print will look like this:

Compatible Version:

image

Not compatible:

image

I sadly don't know which version of GMOD is the current exact minimum on which TTT2 will still run on.

This is a draft, all input welcome :)

(Changelog etc. will follow if accepted and done)

### CODE REVIEW

**Comment by TimGoll** [comment]
I can't tell you the exact minimum. But it should be the February 2024 update. The one wher net.ReadPlayer was added

https://wiki.facepunch.com/gmod/net.ReadPlayer

There is a version linked there. Maybe it helps?

**Comment by NickCloudAT** [comment]
> I can't tell you the exact minimum. But it should be the February 2024 update. The one wher net.ReadPlayer was added
> 
> https://wiki.facepunch.com/gmod/net.ReadPlayer
> 
> There is a version linked there. Maybe it helps?

Ah, yeah that helps. That would be version string "20231006" which I could set of course.

**Comment by TimGoll** [comment]
> > I can't tell you the exact minimum. But it should be the February 2024 update. The one wher net.ReadPlayer was added
> > https://wiki.facepunch.com/gmod/net.ReadPlayer
> > There is a version linked there. Maybe it helps?
> 
> Ah, yeah that helps. That would be version string "20231006" which I could set of course.

Hm. That update was released in the beginning of 2024 though. So the year seems wrong? Maybe it was only the dev branch of that year?

**Comment by NickCloudAT** [comment]
> > > I can't tell you the exact minimum. But it should be the February 2024 update. The one wher net.ReadPlayer was added
> > > https://wiki.facepunch.com/gmod/net.ReadPlayer
> > > There is a version linked there. Maybe it helps?
> > 
> > 
> > Ah, yeah that helps. That would be version string "20231006" which I could set of course.
> 
> Hm. That update was released in the beginning of 2024 though. So the year seems wrong? Maybe it was only the dev branch of that year?

Yeah I just noticed.. I would guess it's the date of first implementation, not public release :/
I mean, theoretically we could just set the current GMOD version as the "baseline".. All servers should be up-to-date anyway.

**Comment by NickCloudAT** [comment]
I guess ready for review.

File: gamemodes/terrortown/gamemode/server/sv_addonchecker.lua [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ADDON_INCOMPATIBLE = 0
 ADDON_OUTDATED = 1
+TTT2_GMOD_MIN_VERSION = 241009
```
Comment (Histalek) [inline gamemodes/terrortown/gamemode/server/sv_addonchecker.lua:9]:
```suggestion
TTT2_GMOD_MIN_VERSION = 240313
```

For better or worse my dev server setup wasn't up to date and this seems to be the march version, which is able to run the current master branch.

**Review by Histalek — APPROVED** [review]
Nice find and cool addition!

It's a pity that we can't get the branch the server is using, but oh well

File: gamemodes/terrortown/gamemode/server/sv_addonchecker.lua (hunk 1 above)
Comment (NickCloudAT) [reply gamemodes/terrortown/gamemode/server/sv_addonchecker.lua]: Ahh, very nice thanks :)

I changed it to that version.

**Pull request merged**